fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LucenePartitioner.java

```diff
@@ -664,9 +669,6 @@ public CompletableFuture<Pair<Integer, Integer>> processPartitionRebalancing(@No
 
                     // get the N oldest documents in the partition (note N = (count of docs to move) + 1, since we need
                     // the (N+1)th doc's timestamp to update the partition's "from" field.
-                    final Integer repartitionDocumentCount = Math.min(
-                            Objects.requireNonNull(state.context.getPropertyStorage().getPropertyValue(LuceneRecordContextProperties.LUCENE_REPARTITION_DOCUMENT_COUNT)),
-                            indexPartitionHighWatermark);
```

Comment:
You dropped the fact that this takes the min of the `indexPartitionHighWatermark` and the provided document count.
It's probably not super important in practice, because your partition high-watermark should be much higher than not, and it allowed `LuceneIndexMaintenanceTest.randomizedRepartitionTest` to be more chaotic (none of the prb tests set the repartition-count higher than the partition high-watermark)

Reply:
Yes, I did think that this check is weird as the high-watermark is much higher. I'll add it for the tests. 
✅